### PR TITLE
Add PTFE Reference Architecture for GCP

### PR DESCRIPTION
**Pull request by roooms**

### CODE REVIEW

File: content/source/docs/enterprise/private/gcp-setup-guide.html.md [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+the infrastructure requirements for PTFE range from a single Cloud Compute VM instance
+for demo installations to multiple instances connected to Cloud SQL, Cloud Storage, and an
+external Vault cluster for a stateless production installation.
+
+This reference architecture focuses on the “Production - External
+Services” operational mode.
+
+The following table provides high-level server guidelines. Of particular
+note is the strong recommendation to avoid non-fixed performance CPUs,
+or “Burstable CPU” in GCP terms, such as T-series instances.
```
Comment (blalor) [inline content/source/docs/enterprise/private/gcp-setup-guide.html.md]: Did GCP adopt instance series naming like AWS?  Do they have an equivalent to `t2.micro`?

File: content/source/docs/enterprise/private/gcp-setup-guide.html.md [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+specified during the UI-based installation or the path to the
+certificate codified during an unattended installation.
+
+If a Classic or Application Load Balancer is used, SSL/TLS will be terminated there.
+In this configuration, the PTFE instances should still be configured to listen
+for incoming SSL/TLS connections but can do so using a self-signed certificate.
+
+Since the load balancer is configured to send traffic to port 443 on the instances
+as type `https`, will ignore the fact that it can't trust the self-signed certificate
+and operate correctly.
```
Comment (blalor) [inline content/source/docs/enterprise/private/gcp-setup-guide.html.md]: Should this be "_we_ will ignore the fact"?

File: content/source/docs/enterprise/private/gcp-setup-guide.html.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+![gcp-infrastructure-diagram](./assets/gcp-infrastructure-diagram.png)
+
+The above diagram shows the infrastructure components at a high-level.
+
+### Application Layer
+
+The Application Layer is composed of two PTFE servers (Compute Engine instances)
+running in different Zones and operating in a main/standby
+configuration. Traffic is routed only to *PTFE-main* via an Alias IP.
+Routing changes are typical managed by a human removing the Alias IP from *PTFE-main*
```
Comment (blalor) [inline content/source/docs/enterprise/private/gcp-setup-guide.html.md]: "are typical" → "are typically"

File: content/source/docs/enterprise/private/gcp-setup-guide.html.md (hunk 1 above)
Comment (roooms) [reply content/source/docs/enterprise/private/gcp-setup-guide.html.md]: They have F and G series instances for bursting. I need to update this.

File: content/source/docs/enterprise/private/gcp-setup-guide.html.md (hunk 2 above)
Comment (roooms) [reply content/source/docs/enterprise/private/gcp-setup-guide.html.md]: I think Evan added this paragraph to a different cloud setup guide. I'll check with him on exactly what this should be.

File: content/source/docs/enterprise/private/gcp-setup-guide.html.md (hunk 3 above)
Comment (roooms) [reply content/source/docs/enterprise/private/gcp-setup-guide.html.md]: Fixed in latest commit

File: content/source/docs/enterprise/private/gcp-setup-guide.html.md (hunk 2 above)
Comment (roooms) [reply content/source/docs/enterprise/private/gcp-setup-guide.html.md]: Updated to match Mary's new wording in the AWS guide

File: content/source/docs/enterprise/private/gcp-setup-guide.html.md [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+| Minimum     | 2-4 core | 8-16 GB RAM  | 50GB | n1-standard-2, n1-standard-4   |
+| Recommended | 4-8 core | 16-32 GB RAM | 50GB | n1-standard-4, n1-standard-8   |
+
+#### Hardware Sizing Considerations
+
+- The minimum size would be appropriate for most initial production
+- deployments, or for development/testing environments.
+
+- The recommended size is for production environments where there is a
+- consistent high workload in the form of concurrent terraform runs.
```
[the remark this answers is not in the document]
Comment (roooms) [reply content/source/docs/enterprise/private/gcp-setup-guide.html.md]: Fixed in latest commit

File: content/source/docs/enterprise/private/gcp-setup-guide.html.md (hunk 1 above)
Comment (roooms) [reply content/source/docs/enterprise/private/gcp-setup-guide.html.md]: Fixed in latest commit

**Comment by roooms** [comment]
@blalor any ideas when this will be merged?

**Comment by roooms** [comment]
@phinze and @topfunky is there anything else I need to do before this can be merged? @blalor is happy with it but Travis failed one build/test. I'm not sure if this is safe to merge.

**Comment by roooms** [comment]
I have another PR (#390) that is somewhat related. Both can be merged separately, but once #390 is merged I will need to rename the page titles and navigation items in this PR as well.

**Comment by phinze** [comment]
Travis checks for broken links so I’d scroll to the bottom and see if
that’s what it caught.

**Comment by roooms** [comment]
Thanks for the pointer @phinze , I didn't know I could see this. Travis shows:

```
Found 3 broken links.
http://127.0.0.1:4567/docs/providers/runscope/index.html
http://127.0.0.1:4567/docs/providers/telefonicaopencloud/index.html
http://127.0.0.1:4567/docs/providers/azurestack/index.html
```

Which I am guessing means some provider links are broken, but I can't find those references in any file in the project.

**Comment by phinze** [comment]
Ah yep those are links in submodules that don't have anything to do with
this PR. Good to ignore!

**Pull request merged**